src/proxy/pool.rs

```diff
@@ -71,18 +71,41 @@ struct PoolState {
     // This is merely a counter to track the overall number of conns this pool spawns
     // to ensure we get unique poolkeys-per-new-conn, it is not a limit
     pool_global_conn_count: AtomicI32,
-    spawner: ConnSpawner,
+    pub spawner: ConnSpawner,
```

Comment:
Need to hoist this out to keep member private. Any ideas other than a spawner() method?